src/richie/apps/courses/models/course.py

```diff
@@ -213,6 +214,28 @@ def state(self):
         return "is_closed"
 
 
+class CoursePluginModel(PagePluginMixin, CMSPlugin):
+    """
+    Course plugin model handles the relation from CoursePlugin
+    to their Course instance
+    """
+
+    page = models.ForeignKey(
+        Page,
+        related_name="course_plugins",
+        limit_choices_to={"publisher_is_draft": True, "course__isnull": False},
```

Comment:
We need to filter out the snapshots here.

There's currently a bug on `master` where snapshots are included in the list of courses on organizations & subjects pages. @sampaccoud corrected this in #423 in [this commit](https://github.com/openfun/richie/pull/423/commits/6362e03a7b25c9f95fea0a3ac9dfc3f9d16ccab9).

We need to do something analogous here, otherwise the snapshots will appear in the dropdown to select a course.

Reply:
You should just need to add a clause:
```python
"node__parent__cms_pages__course__isnull": True
```
But before you do that, try writing a test that surfaces the issues ;-)

Reply:
All done. This need review.